### PR TITLE
Wallet sdk

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
Add documentation to assist external developers in utilizing the "Wallet SDK" features of the web-sdk, Android SDK and the Swift SDK (for iOS).

## Changes
- Added "Wallet SDK" documentation and code examples.

Please note that the Android Kotlin examples have not been written yet. I would like an initial review before porting the examples to Kotlin. The Swift SDK is not complete, so those sections cannot be written yet.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-provider.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
+.. _wallet-sdk-identity-provider:
+
+======================================
+Getting the list of identity providers
+======================================
+
+The list of identity providers can be retrieved from a Concordium node; however, the provided list lacks the necessary metadata for creating or recovering an identity from an identity provider. Therefore, Concordium hosts a service called the wallet-proxy that serves this information to wallets.
```
Comment (shjortConcordium) [inline source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-provider.rst:7]: @dg-concordium is he using semicolon correctly; I am not high-level enough to verify.

File: source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-provider.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+.. _Identity Provider List:
+
++++++++++++++++++++++++++++++++++++++++++++++++++
+Get list of identity providers and their metadata
++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Here is an example of how the list of identity providers can be retrieved from the wallet-proxy service. The libraries used here to perform the requests are merely examples, but feel free to use the libraries already utilized in your project.
```
Comment (shjortConcordium) [inline source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-provider.rst:15]:
```suggestion
Here is an example of how the list of identity providers can be retrieved from the wallet-proxy service. The libraries used here to perform the requests are merely examples, so feel free to use the libraries already utilized in your project.
```

File: source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-recovery.rst [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+* `Generating an identity recovery request`_
+* `Sending an identity recovery request`_
+
++++++++++++++++++++++++++++++++++++++++
+Generating an identity recovery request
++++++++++++++++++++++++++++++++++++++++
+
+The following example demonstrates the process of generating an identity recovery request. Part of the input to the request is a secret value derived from the seed phrase, and this value should be kept secret in a similar manner as the seed phrase and account signing keys.
+
+In the example below functionality for retrieving the list of identity providers is present. To see an example of how to implement this function, please check out :ref:`Identity Provider List`.
```
Comment (shjortConcordium) [inline source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-recovery.rst:18]:
```suggestion
In the example below, functionality for retrieving the list of identity providers is present. To see an example of how to implement this function, please check out :ref:`Identity Provider List`.
```

File: source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-creation.rst [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+.. tabs::
+
+    .. tab::
+
+        TypeScript (Web)
+
+        .. code-block:: javascript
+
+              import {
```
Comment (shjortConcordium) [inline source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-creation.rst:30]:
```suggestion
            import {
```

File: source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-creation.rst [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+                ConcordiumGRPCWebClient,
+                ConcordiumHdWallet,
+                createIdentityRequestWithKeys,
+                IdentityRequestWithKeysInput,
+                IdObjectRequestV1,
+                Versioned,
+            } from '@concordium/web-sdk';
+
+            // Select the identity provider that is to be used for creating the identity. Here we
+            // choose the first one in the list available from the Concordium wallet-proxy.
```
Comment (shjortConcordium) [inline source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-creation.rst:40]: The font on the comments are very annoying(?) to read. :thinking:

File: source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-creation.rst [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+            // Some global cryptographic properties are required as input. They can be retrieved from
+            // a Concordium node using the gRPC interface. Note that these are not specific for this
+            // identity and therefore can be re-used.
+            const client = new ConcordiumGRPCWebClient(nodeAddress, nodePort);
+            const cryptographicParameters = await client.getCryptographicParameters();
+
+            // Derive the secret key material and randomness from the Concordium wallet.
+            const seedPhrase = 'fence tongue sell large master side flock bronze ice accident what humble bring heart swear record valley party jar caution horn cushion endorse position';
+            const network = 'Testnet'; // Or mainnet, if working on mainnet.
```
Comment (shjortConcordium) [inline source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-creation.rst:56]:
```suggestion
            const network = 'Testnet'; // Or Mainnet, if working on mainnet.
```

File: source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-recovery.rst [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                IdentityProvider,
+                IdentityRecoveryRequestWithKeysInput,
+            } from '@concordium/web-sdk';
+
+            // The identity provider to attempt to recover an identity from. Here we simply select the first available, but
+            // in a recovery process a wallet would usually loop through all possible options.
+            const identityProvider: IdentityProvider = getIdentityProviders(walletProxyTestnetBaseUrl)[0];
+
+            const seedPhrase = 'fence tongue sell large master side flock bronze ice accident what humble bring heart swear record valley party jar caution horn cushion endorse position';
+            const network = 'Testnet'; // Or mainnet, if working on mainnet.
```
Comment (shjortConcordium) [inline source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-recovery.rst:41]:
```suggestion
            const network = 'Testnet'; // Or Mainnet, if working on mainnet.
```

File: source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-creation.rst (hunk 5 above)
Comment (orhoj) [reply source/mainnet/net/guides/wallet-sdk/wallet-sdk-identity-creation.rst:40]: I don't know if that can be changed, but I also don't think it is that bad.

**Comment by dg-concordium** [comment]
I made changes directly in the branch for grammar and readability. Before publishing, I'd like the Android code snippets to be there. Is that possible? @orhoj

**Review by dg-concordium — APPROVED** [review]
Once Android examples are in, this is ready to deploy.

**Comment by orhoj** [comment]
Android examples have now been added.

**Pull request merged**